app/services/account_reset/delete_account.rb

```diff
@@ -13,12 +13,11 @@ def call
       track_account_age
       track_mfa_method_counts
 
-      if success
-        notify_user_via_email_of_deletion
-        destroy_user
-      end
+      extra = extra_analytics_attributes
+
+      handle_successful_submission if success
```

Comment:
We need to capture the email addresses before destroying the user so we can cycle through them.